### ISSUE TITLE
Asking for feature: just add commit count behind the version tag (ignoring the "dev" prefix and commit hash)?

### ISSUE THREAD

**Issue by Oliver-Loeffler** [ofek/hatch-vcs#30]
Happy New Year!
Thanks for providing this very nice hatch plugin. Its really a very helpful tool.

When hatch-vcs detects a suitable git tag such as `0.1` which is followed by 3 commits, it will produce a version string such as `0.1.dev3+<commithash>`.

Would it be possible to implement a configuration option so that instead of `0.1.dev3+<commithash>` a version string like `0.1.3` is produced? The commit count in this case shall be used as the patch version.

The background is, I am releasing a small library continually and just migrated to hatch.

Thanks and a great 2023!

**Comment by RonnyPfannschmidt** [comment]
For what reason do you want to do that (people end up with broken artifacts that way all too often

There are multiple ways to enable this, but it needs to be tailored

**Comment by ofek** [comment]
Happy New Year!

I'd recommend opening a feature request to https://github.com/pypa/setuptools_scm since this plug-in mostly just wraps that. FYI @RonnyPfannschmidt is the core maintainer of that 🙂

**Comment by Oliver-Loeffler** [comment]
Thanks. Yes, I figured out that it basically wraps the corresponding setuptools_scm functionality. Thanks for pointing out the contact, I will rethink this and possibly consider a feature request.

**Comment by RonnyPfannschmidt** [comment]
@Oliver-Loeffler depending on the use-case there are reasonable hack to employ for certain use-cases (suitable ci plugins/gh actions impending

if you elaborate i may be able to give a reasonable starting point

**Comment by Oliver-Loeffler** [comment]
Well, the scenario is the following:
* version is read from Git tag (tags are usually formatted like: `0.1` (major.minor)
* tags are defined manually by the developer
* full version is supposed to be like: `0.0.1` (major.minor.patch)
* each commit goes into a CI/CD pipeline
* the actual version number to be build is created like:  (tag.commitcount) which looks like `0.1.21` eventually, the commit count is our patch version

In short, the patch version is the commit count to the last tag.
For local builds, outside CI/CD the pattern `0.1.dev21-hash` is actually perfect.
As each commit progresses into CI/CD, the commit count as patch version is the favoured way for continuously integrated builds.
